### PR TITLE
Added 1D kinematic state message.

### PR DESCRIPTION
**Pull request by jrutgeer**
# 🎉 New feature

Required for https://github.com/gazebosim/gz-sim/pull/2021

## Summary
This PR adds a message type for 1D kinematic state:

- Position
- Velocity
- Acceleration
- Jerk

### CODE REVIEW

File: proto/gz/msgs/kinematic_state_1D.proto [hunk 1]
```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright (C) 2019 Open Source Robotics Foundation
```
Comment (ahcorde) [inline proto/gz/msgs/kinematic_state_1D.proto:2]:
```suggestion
 * Copyright (C) 2023 Open Source Robotics Foundation
```

**Comment by mjcarroll** [comment]
@azeey and @scpeters any other input from a physics perspective?

**Comment by azeey** [comment]
Looking at https://github.com/gazebosim/gz-sim/pull/2021, I only see position and velocity being populated. Have you considered using the existing `Odometery` message?

**Comment by jrutgeer** [comment]
@azeey Good question. The initial idea for https://github.com/gazebosim/gz-sim/pull/2021 was that it would publish the full kinematic state, including acceleration and jerk. However, during implementation I realized that this is not possible as the TrackController is velocity resolved (i.e. at each timestep the position and velocity are known, but the acceleration depends on the next velocity setpoint, which is not yet known).

So technically it is possible to use the odometry message instead, at least for this usecase.

From a semantic point of view, I think initially it feels rather strange to use the odometry message, but having given it some thought I conclude that "conveyor position and velocity" is indeed a kind of "1D odometry", and similar arguments as in [this discussion](https://github.com/ros2/common_interfaces/issues/179#issuecomment-1075676029) hold, so it might indeed make sense to represent a 1D odometry with the existing 6D odometry message.

If you follow this line of thought, I will close this PR and adapt [gazebosim/gz-sim#2021](https://github.com/gazebosim/gz-sim/pull/2021) accordingly.

**Comment by jrutgeer** [comment]
Closing in favor of changing https://github.com/gazebosim/gz-sim/pull/2021 to use an odometry message instead.

**Pull request closed without merging**